### PR TITLE
Corrected the regex for parsing links and some minor fixes

### PR DESCRIPTION
**Pull request by mervick**
**Main fix:**
- fix regex for parsing links e.g. `<a class="..." href="url">...`

**Additional:**

- remove duplicated links
- ignore mailto links
- catch and ignore getRangeAt errors

### CODE REVIEW

File: src/app/utils/selection.ts [hunk 1]
```diff
@@ -45,26 +48,32 @@ function extractLinks(rawHtml: string): string[] {
     // [\n\r\s] - any whitespace character, including linebreaks
     // *? - matches any amount, lazily
     // (?!#) - makes sure that # anchors are not matched - it's unlikely that these would be desired
-    const regexp = /<[\n\r\s]*?a[\n\r\s]*?href[\n\r\s]*?=[\n\r\s]*?"[\n\r\s]*?(?<url>(?!#).*?)[\n\r\s]*?"[\S\s]*?>[\S\s]*?<[\n\r\s]*?\/[\n\r\s]*?a[\n\r\s]*?>/g;
-    
+    // const regexp = /<[\n\r\s]*?a[\n\r\s]*?href[\n\r\s]*?=[\n\r\s]*?"[\n\r\s]*?(?<url>(?!#).*?)[\n\r\s]*?"[\S\s]*?>[\S\s]*?<[\n\r\s]*?\/[\n\r\s]*?a[\n\r\s]*?>/g;
```
Comment (marcinjahn) [inline src/app/utils/selection.ts]: You can remove this whole comment block.

File: src/app/utils/selection.ts [hunk 2]
```diff
@@ -45,26 +48,32 @@ function extractLinks(rawHtml: string): string[] {
...
     return links.map(link => {
         if (link.startsWith("/")) {
             return `${origin}${link}`;
-        } else if (link.startsWith(".")) {
+        }
+        else if (link.startsWith(".")) {
             return `${origin}/${link}`;
         }
-        
+        else if (link.indexOf("mailto:") === 0) {
```
Comment (marcinjahn) [inline src/app/utils/selection.ts]: Good idea to reject `mailto`. However, I think this function is not the right place to add such a guard. I think the regex itself could handle that:

```ts
/<a\s+(?:[^>]*?[\s"']+)?href=(["'])(?!(?:mailto:))(?<url>.*?)\1/gim
```

**Review by marcinjahn — CHANGES_REQUESTED** [review]
Thanks for your contribution, these are nice improvements. I added some small remarks.

File: src/app/utils/selection.ts (hunk 1 above)
Comment (mervick) [reply src/app/utils/selection.ts]: Removed

File: src/app/utils/selection.ts (hunk 2 above)
Comment (mervick) [reply src/app/utils/selection.ts]: Moved to regex

**Review by marcinjahn — CHANGES_REQUESTED** [review]
Please revert the formatting changes. The rest of the codebase is 4 tabs and single quotes.

**Comment by mervick** [comment]
Sorry for the formatting. My local git settings automatically reformat the code.
 Now fixed that

**Review by marcinjahn — APPROVED** [review]

**Comment by marcinjahn** [comment]
I've submitted the new version (1.1.0) to both Google Chrome Web Store and Microsoft Edge Addons. It should be live soon. Thanks again!

**Pull request merged**